lib/internal/Magento/Framework/App/Router/ActionList.php

```diff
@@ -91,7 +91,7 @@ public function get($module, $area, $namespace, $action)
         if ($area) {
             $area = '\\' . $area;
         }
-        if (strpos($namespace, self::NOT_ALLOWED_IN_NAMESPACE_PATH) !== false) {
+        if (strpos(strtolower($namespace), self::NOT_ALLOWED_IN_NAMESPACE_PATH) !== false) {
```

Comment:
Simply use `stripos`. Please cover this case with automated test.

After changes are applied and all builds are green, please squash them into a single commit so that we have perfectly clean history 😉